code/starfield/starfield.cpp

```diff
@@ -1584,13 +1545,13 @@ void stars_draw_bitmaps(int show_bitmaps)
 
 		if (Starfield_bitmaps[star_index].xparent) {
 			if (Starfield_bitmaps[star_index].fps) {
-				bitmap_id = stars_get_next_frame(Starfield_bitmap_instances[idx], Starfield_bitmaps[star_index], timestamp());
+				bitmap_id = Starfield_bitmaps[star_index].bitmap_id + ((timestamp() * Starfield_bitmaps[star_index].fps / MILLISECONDS_PER_SECOND) % Starfield_bitmaps[star_index].n_frames);
 			} else {
 				bitmap_id = Starfield_bitmaps[star_index].bitmap_id;
 			}
 		} else {
 			if (Starfield_bitmaps[star_index].fps) {
-				bitmap_id = stars_get_next_frame(Starfield_bitmap_instances[idx], Starfield_bitmaps[star_index], timestamp());
+				bitmap_id = Starfield_bitmaps[star_index].bitmap_id + ((timestamp() * Starfield_bitmaps[star_index].fps / MILLISECONDS_PER_SECOND) % Starfield_bitmaps[star_index].n_frames);
```

Comment:
An int being divided by an int, this is integer division so if this is 30 fps / 1000 won't it always be 0?

Reply:
The error was a unit mismatch, it's been corrected now and tested to function correctly. After this change, the output is corrected via:
```
timestamp() * Starfield_bitmaps[star_index].fps / MILLISECONDS_PER_SECOND
=
milliseconds * (frames / seconds) / 1000
=
seconds * 1000 * (frames / seconds) / 1000
= 
frames
```

Reply:
`seconds * 1000 * (frames / seconds) / 1000` this is the part that worries me, since it's unclear what the order of operations is, and it would change the result.
I trust you that it works out, but I would appreciate parentheses `((timestamp() * Starfield_bitmaps[star_index].fps) / MILLISECONDS_PER_SECOND)` to make it more clear that this operation is executing as it should.

Reply:
This should be addressed now.